GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -884,9 +884,11 @@ public void ForceRefreshRevisions()
 
                 SelectInitialRevision(newCurrentCheckout);
 
-                // NB: Build ref filter before updating the property to reduce the number of chage events.
+                const RefFilterOptions gitNotesOptions = RefFilterOptions.All | RefFilterOptions.Boundary;
+
+                // NB: Build ref filter before updating the property to reduce the number of change events.
                 RefFilterOptions refFilterOptions = RefFilterOptions;
-                if (!AppSettings.ShowGitNotes && refFilterOptions.HasFlag(RefFilterOptions.All | RefFilterOptions.Boundary))
+                if (!AppSettings.ShowGitNotes && (refFilterOptions & gitNotesOptions) == gitNotesOptions)
```

Comment:
IIRC .NET Core has resolved this allocations, hasn't it? https://github.com/gitextensions/gitextensions/pull/8522 makes this change obsolete.

Reply:
Yes, it's not needed under .NET Core. What is the timeline for porting?

Until such a time, I suggest merging this. I ran a trace over GE startup and this was a quick win from the top 30 types allocated on the heap.

Reply:
then this should go into 3.4 too

Reply:
Sibling PR targeting 3.5: #8889